node/podcontroller.go

```diff
@@ -304,6 +305,26 @@ func (pc *PodController) Run(ctx context.Context, podSyncWorkers int) (retErr er
 			if key, err := cache.MetaNamespaceKeyFunc(newPod); err != nil {
 				log.G(ctx).Error(err)
 			} else {
+				obj, ok := pc.knownPods.Load(key)
+				if !ok {
+					// Pods are only ever *added* to knownPods in the above AddFunc, and removed
+					// in the below *DeleteFunc*
+					panic("Pod not found in known pods. This should never happen.")
```

Comment:
Seems a bit extreme. What's the expected result by doing this? VK crashing?

Reply:
Yeah, this indicates a critical bug / internal data state corruption.

Reply:
Gotcha!